service/header/p2p_subscriber.go

```diff
@@ -78,3 +80,36 @@ func (p *P2PSubscriber) Broadcast(ctx context.Context, header *ExtendedHeader) e
 	}
 	return p.topic.Publish(ctx, bin)
 }
+
+// msgID computes an id for a pubsub message
+// TODO(@Wondertan): This cause additional allocations per each recvd message in the topic. Find a way to avoid those.
+func msgID(pmsg *pb.Message) string {
+	msgID := func(data []byte) string {
+		hash := blake2b.Sum256(data)
+		return string(hash[:])
+	}
+
+	h, err := UnmarshalExtendedHeader(pmsg.Data)
+	if err != nil {
+		log.Errorw("unmarshalling header while computing msg id", "err", err)
+		return msgID(pmsg.Data)
```

Comment:
So i guess technically if we run into some marshalling errs in this msgID function, then we *could* have duplicated header messages in the network, but it's unlikely right?

Reply:
ack